### DISCUSSION TITLE
Can I use CRA to make a library in React?

### DISCUSSION THREAD

**Discussion by vinimarcili** [dilanx/craco#485] [Q&A]
If I have a project using CRA can I somehow reconfigure webpack so it doesn't need an index.html and just export the components to be published to NPM? I couldn't find this anywhere, I just want to know if it's possible and if so how to do it

**Comment by dilanx** [comment]
You shouldn't need webpack to publish to npm. You can use the [`files`](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#files) property in your `package.json` to specify which files you want published (like your components).

**Reply by vinimarcili** [reply]
Yes I know that, but I need to go through some builds (scoped scss, ts-paths) and I can with craco. That's why I asked the question because I dont need the index.html when I build a library

**Reply by dilanx** [reply]
I see. That would be more of a webpack question than craco, since craco can modify the webpack config to be whatever you'd like. Usually, I wouldn't use CRA to make a library since its purpose is to make SPAs, and would instead use TypeScript's compiler plus any other preprocessing commands (in package.json and/or as a CI workflow or something) to build the library.

Another option could be to exclude the html files produced by webpack when publishing.

**Reply by vinimarcili** [reply]
Ok, thanks for the help!
I'll try one of the alternatives you mentioned!
Thanks
